lib/container.ex

```diff
@@ -2,12 +2,8 @@
 # Original by: Marco Dallagiacoma @ 2023 in https://github.com/dallagi/excontainers
 # Modified by: Jarl André Hübenthal @ 2023
 defmodule Testcontainers.Container do
```

Comment:
this refactor is isolating the Container modules purpose to just being a struct with builder functions